packages/ContentAutotagging/Engine/generic/AutotagBaseEngine.ts

```diff
@@ -0,0 +1,745 @@
+            Please provide the keywords in a list format.
+            Make sure the response is just the json file without and formatting or code blocks, and strictly following the format below. Please don't include a greeting in the response, only output the json file:
+
+            {
+                "title": (title here),
+                "keywords": (list keywords here), 
+                "isValidContent": true (as a boolean)
+            }
+
+            ${additionalContentTypePrompts}
+
+            Please make sure the response in is valid JSON format. 
+
+            The supplied text is: ${chunk}
+            `;
+
+            const response = await LLM.ChatCompletion({
+                messages: [
+                    {
+                        role: 'system',
+                        content: systemPrompt,
+                    },
+                    {
+                        role: 'user',
+                        content: userPrompt,
+                    }
+                ],
+                model: model,
+                temperature:0.0, 
+                response_format: {"type":"json_object"}
+            });
+
+            const queryResponse = response.data.choices[0]?.message?.content?.trim() || '';
+            const JSONQueryResponse: JsonObject = JSON.parse(queryResponse);
+
+            // check if the response has info to add to LLMResults
+            for (const key in JSONQueryResponse) {
+                const value = JSONQueryResponse[key]
+                if (value !== null) { 
+                    LLMResults[key] = value
+                }
+            }
+        }   
+        
+        const endTime = new Date();
+        LLMResults.processStartTime = startTime
+        LLMResults.processEndTime = endTime
+        LLMResults.contentItemID = params.contentItemID
+
+        return LLMResults
+    }
+
+    public async saveLLMResults(LLMResults: JsonObject, contextUser: UserInfo) {
+        if (LLMResults.isValidContent === true) {   
+            // Only save results if the content is of the type that we expected. 
+            await this.saveResultsToContentItemAttribute(LLMResults, contextUser)
+            await this.saveContentItemTags(LLMResults.contentItemID, LLMResults, contextUser)
+            console.log(`Results for content item ${LLMResults.contentItemID} saved successfully`)
+        }
+        else {
+            await this.deleteInvalidContentItem(LLMResults.contentItemID, contextUser)
+        }
+    }
+
+    public async deleteInvalidContentItem(contentItemID: string, contextUser: UserInfo) {
+        const md = new Metadata()
+        const contentItem: ContentItemEntity = <ContentItemEntity> await md.GetEntityObject('Content Items', contextUser)
+        await contentItem.Load(contentItemID)
+        await contentItem.Delete()
+    }
+
+    public chunkExtractedText(text: string, modelAPIName: string): string[]{
```

Comment:
For the next version of this functionality we need to enable smarter chunking using something like the tool I shared on Teams. Let's build a generic way to do this by having a "chunking" provider base and implementation sub-classes and maybe some metadata (might not be needed) to make various chunking strategies available and then pick the highest priority implementation of the base class dynamically at run time. And then call that from here. Add an issue for this so you can track and implement after your return please.

@nico-oz